### PR TITLE
Care gaps request

### PR DESCRIPTION
**Pull request by sakhin123**
## Summary
Implementation of Care Gaps request preview, calculate button, and server request/response handling

## New behavior
- The "Calculate" button on the care-gaps page is conditionally enabled depending on whether the user has inputted the necessary inputs for a valid care gaps request.
- The request preview is now implemented - user's inputted values included in the preview as the user inputs them
- If the "Calculate" button is pressed, the request built in the preview is sent to the server for processing
- If the request is successful, a success notification is displayed for the user and the care gaps response is displayed in a Prism component
- If unsuccessful, an error notification is displayed

## Code changes
- **pages/[resource]/[id]/care-gaps.tsx** implements request preview and request sending 
- **__tests__/pages/resource/id/care-gaps.tsx** includes unit tests for new functionality
- **__tests__/pages/resource/id/evaluate.tsx** altered so unit tests use DateTime.now().year instead of hard-coded 2022

## Testing Guidance
- See the README.md for first time setup instructions.
#### Test in a browser
- To start the frontend: `npm run dev` then navigate to http://localhost:3001 in a browser
1. Valid navigation:
    - Click on “Measure” from the left-hand-side navigation menu. Click on any of the measure IDs. Click the “Calculate Care Gaps” button
    - The request preview and calculate button should be rendered (no "Placeholder" labels)
1. Successful care-gaps evaluation:
    - Choose a valid set of options to build the care-gaps request:
      a. Choose reportType: Subject, then select a Patient  
               OR
      b. Chose reportType: Organization, then select an Organization
    - Click the Calculate button. A success notification and a Prism component with the care-gaps report should both render
2. Failed care-gaps evaluation:
    - Create a new Measure resource with no body (from the Measure home, using the Create button).
      - Use: ` {"resourceType": "Measure"}`
    - On the care-gaps page for that new measure, click the Calculate button and an error notification should appear, specifying the issue.

#### Unit tests
- Run the unit tests: `npm run test`

### CODE REVIEW

File: __tests__/pages/resource/id/evaluate.test.tsx [hunk 1]
```diff
@@ -322,7 +327,9 @@ describe("Select component, Radio button, and request preview render", () => {
     //expect the request preview to include reportType=population
     expect(
       screen.getByText(
-        "/Measure/Measure-12/$evaluate-measure?periodStart=2022-01-01&periodEnd=2022-12-31&reportType=population",
+        `/Measure/Measure-12/$evaluate-measure?periodStart=${DateTime.now().year}-01-01&periodEnd=${
```
Comment (mgramigna) [inline __tests__/pages/resource/id/evaluate.test.tsx:330]: We now have you to thank for when the unit tests don't mysteriously break in 2023 lol

File: pages/[resourceType]/[id]/care-gaps.tsx [hunk 2]
```diff
@@ -57,6 +61,116 @@ const CareGapsPage = () => {
...
+    if (
+      (periodStart && periodEnd && radioValue === "Subject" && patientValue) ||
+      (periodStart && periodEnd && radioValue === "Organization" && organizationValue)
+    ) {
+      return true;
+    } else return false;
+  };
+
+  //handles sending the care-gaps request and processes the response
+  function calculateCareGapsHandler() {
```
Comment (mgramigna) [inline pages/[resourceType]/[id]/care-gaps.tsx:101]:
```suggestion
  const calculateCareGapsHandler = () => {
```

File: pages/[resourceType]/[id]/care-gaps.tsx [hunk 3]
```diff
@@ -57,6 +61,116 @@ const CareGapsPage = () => {
...
+      icon: <X size={18} />,
+      autoClose: false,
+    };
+    let fetchStatus = { status: 500, statusText: "Failed fetch request" };
+    setLoadingRequest(true);
+
+    fetch(`${process.env.NEXT_PUBLIC_DEQM_SERVER}${createRequestPreview()}`)
+      .then((response) => {
+        fetchStatus = { status: response.status, statusText: response.statusText };
+        return response.json();
```
Comment (mgramigna) [inline pages/[resourceType]/[id]/care-gaps.tsx:115]: Remember to use strong typing/type inference. In fetch requests where we know the return type for certain. In this case, response.json() is a `Promise<fhirJson.Parameters | fhirJson.OperationOutcome>`.

I think I made a similar comment in the `MeasuredatePickers` component, so you can look back at that for guidance

File: pages/[resourceType]/[id]/care-gaps.tsx [hunk 4]
```diff
@@ -262,7 +376,7 @@ const CareGapsPage = () => {
                           colorScheme="dark"
                           style={{ maxWidth: "77vw", height: "80vh", backgroundColor: "#FFFFFF" }}
```
Comment (elsaperelli) [inline pages/[resourceType]/[id]/care-gaps.tsx:377]: image


Not sure if this is even something you would want to change but I figured that it was worth pointing out! I noticed that the care gaps input and prism component don't quite line up if the window is either resized or the request preview is pretty long (i.e. includes both an organization or practitioner). Very nit-picky UI thing I found so very understandable if this isn't a priority!

**Review by elsaperelli — CHANGES_REQUESTED** [review]
Just one UI suggestion! Other than that great work.

File: pages/[resourceType]/[id]/care-gaps.tsx (hunk 4 above)
Comment (sakhin123) [reply pages/[resourceType]/[id]/care-gaps.tsx:380]: I adjusted some padding so the alignment is a little better but it's still not great if the user resizes the page. I'll look into making it more dynamic for this and evaluate measure later in the week once I finish up search pagination stuff, if that's ok?

File: __tests__/pages/resource/id/evaluate.test.tsx (hunk 1 above)
Comment (sakhin123) [reply __tests__/pages/resource/id/evaluate.test.tsx:330]: I realized I forgot to make this fix in MeasureDatePickers.test.tsx so I just added it

**Review by mgramigna — APPROVED** [review]

File: pages/[resourceType]/[id]/care-gaps.tsx (hunk 4 above)
Comment (elsaperelli) [reply pages/[resourceType]/[id]/care-gaps.tsx:380]: Of course! It really isn't a huge deal at all and I don't even mind it I just thought I would point it out.

**Review by elsaperelli — APPROVED** [review]

**Pull request merged**